src/routes/webhooks/paddle.ts

```diff
@@ -66,6 +88,50 @@ const updateUserSubscription = async ({
   );
 };
 
+const logPaddleAnalyticsEvent = async (
+  data:
+    | SubscriptionUpdatedEvent
+    | SubscriptionCanceledEvent
+    | TransactionPaidEvent
+    | undefined,
+  eventName: AnalyticsEventName,
+) => {
+  if (!data) {
+    return;
+  }
+
+  const customData = data.data?.customData as { user_id: string };
+  const cycle = extractSubscriptionType(data.data?.items);
+  const cost = data.data?.items?.[0]?.price?.unitPrice?.amount;
+  const currency = data.data?.items?.[0]?.price?.unitPrice?.currencyCode;
+  const userId = customData?.user_id || data?.data.id || '';
```

Comment:
We must make sure we use the actual user id with no fallbacks. This is critical for every event.

We can consider fetching the user id by subscription id, if it's missing from the webhook

Reply:
Oh just had an idea, we can query our own db by subscription ID i guess 🤩